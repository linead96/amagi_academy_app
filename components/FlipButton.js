import React from 'react';
import {
  View,
  TouchableNativeFeedback,
  Text,
  StyleSheet,
  } from 'react-native'

const FlipButton = (props) => {
  return (
    <View style={styles.flipButtonContainer}>
        <TouchableNativeFeedback onPress={() => props.flip()}>
            <View style={styles.flipButton}>
                <Text style={styles.flipButton__text}>Flip</Text>
            </View>
        </TouchableNativeFeedback>  
    </View>
  )
}

const styles = StyleSheet.create({
    flipButtonContainer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        width: '30%',
    },
    flipButton: {
        backgroundColor: 'yellow',  
        flex: 1,
        width: '100%',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 25,
        marginBottom: '5%',
    },
    flipButton__text: {
        textAlign: 'center',
        fontSize: 30,
        margin: 10,
    },
});

export default FlipButton;