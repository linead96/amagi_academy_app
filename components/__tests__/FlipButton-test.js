import 'react-native';
import React from 'react';
import FlipButton from '../FlipButton';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<FlipButton flipped={false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});