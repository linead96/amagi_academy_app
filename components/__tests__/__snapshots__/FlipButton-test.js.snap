// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "width": "30%",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffbcbc",
          "borderColor": "red",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "margin": 10,
          "width": 300,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Object {
          "color": "#333333",
          "margin": 20,
        }
      }
    >
      TouchableNativeFeedback is not supported on this platform!
    </Text>
  </View>
</View>
`;
